draft-ietf-dnsop-svcb-httpssvc.md

```diff
@@ -790,7 +785,8 @@ All the SvcParamKeys defined in {{keys}} are permitted for use in
 HTTPSSVC, and the "alpn" SvcParamKey is REQUIRED.  Its value MUST
 be an ALPN that is suitable for use with HTTPS.  For example, the
 value MAY be "http/1.1", "h2", or "h3", but MUST NOT be "h2c" or
-"ftp".
+"ftp".  Regardless of the "alpn" SvcParamValue, all service endpoints
+MUST support HTTP/1.1.
 
```

Comment:
What about DoH that mandates H2 support?  It also is entirely reasonable for service endpoints to support QUIC and not H2 or HTTP/1.1.  There are features of QUIC that make it attractive to have it on separate IPs that are QUIC-only.

Special-casing http/1.1 seems reasonable, but I'm not sure this is the easiest way.  
Perhaps:
A service endpoint supporting "h2" SHOULD also support "http/1.1" unless either no service endpoints support "http/1.1" or a distinct service endpoint is specified for "http/1.1"

?

Reply:
> What about DoH that mandates H2 support?

DoH doesn't recommend refusing connections over HTTP/1.1, so I don't think there's a conflict.

> There are features of QUIC that make it attractive to have it on separate IPs that are QUIC-only.

I'd be really interested to know more about this use case.  I've been hearing concerns from browser people that support for QUIC-only endpoints may be difficult to implement, because it complicates TCP fallback when UDP is failing.  I'm hoping that we can find a middle ground that addresses both needs.  (Feel free to switch to email if it's too long for this comment thread.)